### PR TITLE
What is Self-Supervised Learning?

### LINKED ISSUE #1740: What is Self-Supervised Learning?
Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

### Proposed title of article
Your title should be descriptive of the article/tutorial. Be Specific. *Use keyword research to gain your article higher ranking.*

What is Self-Supervised Learning?

### Introduction paragraph (2-3 paragraphs):

How do humans learn so quickly?

Humans learn by using both supervised learning and unsupervised learning. We are multifaceted. This multifaced approach to learning has been so easy for humans. But, that has not been the case for deep learning systems. Though we have seen robust deep learning systems that have performed image recognition and natural language processing tasks, performing complex tasks has remained challenging. This is the problem self-supervised learning is trying to address.

### Key takeaways: 
What are the 3-5 most important things the reader should understand or be able to do after reading this article? Use this area to get your ideas down on the bulk of your article or tutorial.

By the end of this article, the reader should be able to understand:

1. What is Supervised Learning?
2. The challenges associated with Supervised Learning
3. Introduction to Self-Supervised Learning
4. Applications of Self-Supervised Learning

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

1. [A Simple Framework for Contrastive Learning of Visual Representations](https://arxiv.org/pdf/2002.05709.pdf)
2. [Unsupervised Representation Learning by Predicting Image Rotations](https://arxiv.org/pdf/1803.07728.pdf)
3. [Generative Pre-trained Transformer 3 (GPT-3)](https://openai.com/blog/openai-api/)
4. [Bidirectional Encoder Representations from Transformers (BERT)](https://arxiv.org/pdf/1810.04805.pdf)

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by Bayler**
This Fixes #1740

### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

File: articles/what-is-self-supervised-learning/index.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+4. [Challenges with Supervised Learning](#challenges-with-supervised-learning)
+5. [Introduction to Self-Supervised Learning](#introduction-to-self-supervised-learning)
+6. [Applications of Self-Supervised Learning](#applications-of-self-supervised-learning)
+7. [Summary](#summary)
+7. [Wrapping Up](#wrapping-up)
+8. [References](#references)
+
+### Prerequisites
+
+Before reading this article, the reader needs to have little knowledge of Artificial Intelligence and Machine Learning. If you're still a beginner, feel free to read this [article](/section-engineering-education/supervised-learning-algorithms/) published on [Section](https://www.section.io/engineering-education/) to understand the differences between supervised learning or unsupervised learning on a larger scale.
```
Comment (collins-a) [inline articles/what-is-self-supervised-learning/index.md:34]:
```suggestion
Before reading this article, a reader should have basic knowledge of Artificial Intelligence and Machine Learning. If you're a beginner, feel free to read this [article](/section-engineering-education/supervised-learning-algorithms/) published on [Section](https://www.section.io/engineering-education/) to better understand the differences between supervised learning and unsupervised learning.
```

File: articles/what-is-self-supervised-learning/index.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+7. [Wrapping Up](#wrapping-up)
+8. [References](#references)
+
+### Prerequisites
+
+Before reading this article, the reader needs to have little knowledge of Artificial Intelligence and Machine Learning. If you're still a beginner, feel free to read this [article](/section-engineering-education/supervised-learning-algorithms/) published on [Section](https://www.section.io/engineering-education/) to understand the differences between supervised learning or unsupervised learning on a larger scale. 
+
+### Introduction
+
+Artificial Intelligence (AI) is a rapidly advancing field with various techniques being billed as the most promising to keep an eye on. One of those techniques is self-supervised learning. You'll want to read this article to the end to learn more about it.
```
Comment (collins-a) [inline articles/what-is-self-supervised-learning/index.md:38]:
```suggestion
Artificial Intelligence (AI) is a rapidly advancing field with various techniques being billed as the most promising to keep an eye on. One of those techniques is self-supervised learning. This article introduces self-supervised learning.
```

File: articles/what-is-self-supervised-learning/index.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+### What is Supervised Learning?
+
+Supervised learning is a useful technique in deep learning. It is the most widely used type of learning when it comes to AI. It is a learning type where machines learn from labeled data to perform tasks such as predicting and classification of data. For example, a supervisor (someone who knows how to write answers) points out mistakes to a student during the learning process. The student ends up correcting the mistakes until he/she becomes good at them.
```
Comment (collins-a) [inline articles/what-is-self-supervised-learning/index.md:42]: Could you paraphrase the meaning of 'supervisor' in this context? Or add a few words, as it seems a bit ambiguous as is.

File: articles/what-is-self-supervised-learning/index.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+In the supervised learning setting, we want an AI system to consider some data, i.e., given an animal's image, it can classify it with a label. For example, an image of a dog can be classified with the label "a dog." Once the AI system has been trained on the correct labeling of different animals with their correct labels, it should classify new unseen animal data with their correct labels.
+
+Supervised learning is how banks know whether you are qualified to be given a loan or not, how Facebook tells apart your face from your friends face, and how email companies can classify your emails into different categories. But, supervised learning does have its drawbacks.
```
Comment (collins-a) [inline articles/what-is-self-supervised-learning/index.md:46]:
```suggestion
Supervised learning is how banks know whether you are qualified to be given a loan or not, how Facebook tells apart your face from your friend's face, and how email companies can classify your emails into different categories. But, supervised learning does have its drawbacks.
```

File: articles/what-is-self-supervised-learning/index.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+Self-supervised learning has recently been a hot topic in Artificial Intelligence. It is a form of supervised learning that doesn't require human input to perform data labeling. It can also be viewed as a branch of unsupervised learning. The results are obtained by models that analyze data, label and categorize information independently without any human input. The only difference is that, unlike unsupervised learning, self-supervised learning does not perform the grouping and clustering of data, as is the case with unsupervised learning. 
+
+This learning type allows machines to examine part of a data example to figure out the remaining part. In simple terms, self-supervised learning learns from unlabeled data to fill in the blanks for missing pieces. These data can be in the form of images, text, audio, and videos.
```
Comment (collins-a) [inline articles/what-is-self-supervised-learning/index.md:65]:
```suggestion
This learning type allows machines to examine part of a data example to figure out the remaining part. In simple terms, self-supervised learning learns from unlabeled data to fill in the blanks for missing pieces. This data can be in the form of images, text, audio, and videos.
```

File: articles/what-is-self-supervised-learning/index.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+This learning type allows machines to examine part of a data example to figure out the remaining part. In simple terms, self-supervised learning learns from unlabeled data to fill in the blanks for missing pieces. These data can be in the form of images, text, audio, and videos. 
+For example, in videos, the machine can predict the missing part of a video given only a video section. Videos can also use it in predicting missing frames in a video.
+
+Self-supervised learning aims to make deep learning models data-efficient. This means that it helps reduce the over-dependence on vast amounts of data to achieve good models.
+
+### Applications of Self-Supervised Learning
+
+1. Natural Language Processing
```
Comment (collins-a) [inline articles/what-is-self-supervised-learning/index.md:72]:
```suggestion
**1. Natural Language Processing**
```

File: articles/what-is-self-supervised-learning/index.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+Self-supervised learning helps predict the missing words in a text in the field of NLP. This is achieved by showing segments of texts to a giant neural network with billions of parameters, i.e., the likes of OpenAI's [GPT-3](https://openai.com/blog/openai-api/) and Google's [BERT](https://arxiv.org/pdf/1810.04805.pdf). You mask 15% of the text to force the network to predict the pieces of words that are missing. 
+
+2. Computer Vision
```
Comment (collins-a) [inline articles/what-is-self-supervised-learning/index.md:76]:
```suggestion
**2. Computer Vision**
```

File: articles/what-is-self-supervised-learning/index.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+[SimCLR](https://arxiv.org/pdf/2002.05709.pdf) is a framework used to learn visual representations in images using self-supervised learning. The framework performs two main tasks: a pretext task and the downstream (real) task. Self-supervised learning is used in the pretext task. It involves performing simple augmentation tasks such as random cropping, random color distortions, and random Gaussian blur on input images. This process enables the model to learn better representations of the input images. These results are passed into the downstream task module, which performs the main tasks such as detection and classification tasks.
```
Comment (collins-a) [inline articles/what-is-self-supervised-learning/index.md:78]: Perhaps briefly define a random Gaussian blur for a reader who might be unaware.

File: articles/what-is-self-supervised-learning/index.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+### Summary
+
+1. Self-supervised learning exploits unlabeled data to yield labels. This eliminates the need for manually labeling data, which is a tedious process.
+2. They design supervised tasks such as pretext tasks that learn meaningful representation to perform downstream tasks such as detection and classification.
+3. This type of learning helps fill in the blanks. For example, they help predict missing words in NLP.
+
+### Wrapping Up
+
+Self-supervised learning has helped develop AI systems that learn with fewer samples or fewer trials. This has been evident in the Natural Language Processing (NLP) field with GPT-3 and BERT, which can learn with very few examples. This network aims not to learn using labeled data but good representations from unlabeled data. This reduces the need to depend on vast amounts of data like in the supervised learning setting. 
+Currently, self-supervised learning is still in its infant stage. Machines cannot learn and understand almost everything humans can, but it seems to be an exciting and promising step in the right direction.
```
Comment (collins-a) [inline articles/what-is-self-supervised-learning/index.md:89]:
```suggestion
Currently, self-supervised learning is still in its infancy. Machines cannot yet learn or understand everything humans can, but it seems to be an exciting and promising step in the right direction.
```

**Review by collins-a — CHANGES_REQUESTED** [review]
@Bayler good article introducing self-supervised learning. Just a few changes and we'll be ready for the final review.

File: articles/what-is-self-supervised-learning/index.md (hunk 3 above)
Comment (Bayler) [reply articles/what-is-self-supervised-learning/index.md]: I've edited and made it simpler to understand

**Review by collins-a — APPROVED** [review]
Everything seems to be in order

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
@Bayler  made some minor changes - please go over to make sure I didnt omit anything crucial - and to see the changes made for flowl 🚀

**Pull request merged**